BUG [table]: fix a bug where column names would be lost when instanciating Table from a list of Row objects (#15735)
astropy/astropy: docs/whatsnew/6.1.rst

* X distinct people have contributed code

.. _whatsnew-6.1-table-column-order-from-rows:

> Ordering of table columns constructed from ``rows``
> ===================================================
> 
> The order of columns as they appear in a ``Table`` constructed as ``Table(rows=...)``
> may differ between ``astropy`` ``6.0`` and ``6.1``. Specifically, in ``6.0``,
> then column ordering was determined by the first :class:`dict` if it contained values
> for all the columns, or by sorting the column names alphabetically if it did not.
> In ``6.1``, columns are simply inserted as they appear when the :class:`list` of
> :class:`dict` is iterated over.

Reviewer: ```suggestion
Ordering of table columns constructed from rows
===============================================

The column order in a `~astropy.table.Table` constructed from a `list` or rows (`dict`
or `~astropy.table.Row`) may change using ``astropy`` version ``6.1`` *if the 
first row has missing values*.

Before ``6.1``, the column ordering was determined from the first row if it contained
values for all the columns, or by sorting the final column names alphabetically if it 
did not. Starting with ``6.1``, columns are always added in the order they appear 
when iterating over  the `list` of rows.
```

> As an example,

Reviewer: As a general rule, do not make a single long sentence containing multiple code blocks. Instead make a short sentence or clause in front of each one.
```suggestion
For example, create a table as shown below:
```

>     >>> data = [{'b': 10, 'c': 7, },

Reviewer: ```suggestion
    >>> from astropy.table import Table
    >>> data = [{'b': 10, 'c': 7, },
```

    ...         {'a': 15, 'c': 35, 'b': 20}]

>     >>> t = Table(rows=data)

Reviewer: A list of rows as `list[dict | Row]` can be passed as the first argument, and that is probably the most common. The `rows=..` is only needed for a `list[list]` input. But we can highlight that both work.
```suggestion
    >>> t = Table(data)  # or Table(rows=data), which is equivalent
```

>     >>> print(t)

Reviewer: Sphinx testing will fail because this does not print anything.
```suggestion
```

> would print, in ``6.0``

Reviewer: ```suggestion
Before ``6.1`` the table would look like this:
```

     a   b   c
    --- --- ---
     --  10   7
     15  20  35

> and in ``6.1``

Reviewer: ```suggestion
Starting with ``6.1`` the table would instead look like this:
```
